### PR TITLE
Multiple modules: decouple GVL IDs from storage access control

### PR DESCRIPTION
**Pull request by dgirardi**
## Type of change

- [x] Refactoring (no functional changes, no api changes)

## Description of change

This refactors `storageManager` so that it no longer deals with GVL IDs directly. Currently a module can pass it a GVL ID which gets [mostly ignored](https://github.com/prebid/Prebid.js/issues/9717#issuecomment-1487362558); in addition, with the upcoming [activity controls](https://github.com/prebid/Prebid.js/issues/9546) GVL IDs will be just one of the mechanisms that will control access to storage.

With this PR:

- modules must ID themselves fully to `storageManager` (a module type and name are now required);
- GVL IDs are tracked separately; userID will keep track of all userId submodule GVLIDs, core does the same for bid adapters, etc
- RTD modules can also now declare a `gvlid` in their spec in the same vein as all other module types.

### CODE REVIEW

**Review by Fawke — APPROVED** [review]
Hi @dgirardi,

Thanks for the PR and all the the refactoring. All looks good from my end. I agree, that the `gvlid` parameter passed to storageManager wasn't doing much. 

I don't think we'll need to make any docs change for this. I could not find any code snippet or example of initialising storageManager.

File: src/storageManager.js [hunk 1]
```diff
@@ -252,31 +231,38 @@ export function newStorageManager({gvlid, moduleName, bidderCode, moduleType} =
...
+    throw new Error(`Invalid invocation for getStorageManager: must set either bidderCode, or moduleType + moduleName`)
+  }
+  if (bidderCode) {
+    if ((moduleType && moduleType !== MODULE_TYPE_BIDDER) || moduleName) err()
+    moduleType = MODULE_TYPE_BIDDER;
+    moduleName = bidderCode;
+  } else if (!moduleName || !moduleType) {
+    err()
+  }
+  return newStorageManager({moduleType, moduleName});
```
Comment (smenzer) [inline src/storageManager.js:256]: could be a dumb question, but should this line be this?

```js
return newStorageManager({moduleName: moduleName, moduleType: moduleType});
```

**Review by smenzer — COMMENTED** [review]
I focused only on the user id module pieces, but this LGTM

File: src/storageManager.js (hunk 1 above)
Comment (smenzer) [reply src/storageManager.js:256]: got it!

**Pull request merged**